File: frontend/src/types/state.ts
export interface SystemPrompt {
  content: string;
  version: number;
  created_at: string;
}

export interface EvaluationSnapshot {
  id: string;
  email_id: string;
  expected_outcome: string;
  prompt_version: number;
  performance_score: number;
  created_at: string;
}

export interface UserPreference {
  key: string;
  value: string;
  description: string;
  created_at: string;
}

export interface SystemState {
  current_prompt: SystemPrompt;
  user_preferences: UserPreference[];
  evaluation_snapshots: EvaluationSnapshot[];
  optimization_history: Record<string, any>[];
  confidence_score: number;
  last_updated: string;
}

File: backend/app/models/state.py
from typing import List, Dict, Any
from pydantic import BaseModel
from datetime import datetime


class SystemPrompt(BaseModel):
    content: str
    version: int
    created_at: datetime


class EvaluationSnapshot(BaseModel):
    id: str
    email_id: str
    expected_outcome: str
    prompt_version: int
    performance_score: float
    created_at: datetime


class UserPreference(BaseModel):
    key: str
    value: str
    description: str
    created_at: datetime


class SystemState(BaseModel):
    current_prompt: SystemPrompt
    user_preferences: List[UserPreference]
    evaluation_snapshots: List[EvaluationSnapshot]
    optimization_history: List[Dict[str, Any]]
    confidence_score: float
    last_updated: datetime